test/e2e/common/projected_configmap.go

```diff
@@ -0,0 +1,663 @@
+	*/
+	framework.ConformanceIt("should be consumable from pods in volume as non-root [NodeConformance]", func() {
+		doProjectedConfigMapE2EWithoutMappings(f, 1000, 0, nil)
+	})
+
+	It("should be consumable from pods in volume as non-root with FSGroup [NodeFeature:FSGroup]", func() {
+		doProjectedConfigMapE2EWithoutMappings(f, 1000, 1001, nil)
+	})
+
+	/*
+	   Release : v1.9
+	   Testname: Projected Volume, ConfigMap, mapped
+	   Description: A Pod is created with projected volume source ‘ConfigMap’ to store a configMap with default permission mode. The ConfigMap is also mapped to a custom path. Pod MUST be able to read the content of the ConfigMap from the custom location successfully and the mode on the volume MUST be -rw—r——r—-.
+	*/
+	framework.ConformanceIt("should be consumable from pods in volume with mappings [NodeConformance]", func() {
+		doProjectedConfigMapE2EWithMappings(f, 0, 0, nil)
+	})
+
+	/*
+	   Release : v1.9
+	   Testname: Projected Volume, ConfigMap, mapped, volume mode 0400
+	   Description: A Pod is created with projected volume source ‘ConfigMap’ to store a configMap with permission mode set to 0400. The ConfigMap is also mapped to a custom path. Pod MUST be able to read the content of the ConfigMap from the custom location successfully and the mode on the volume MUST be -r-—r——r—-.
+	*/
+	framework.ConformanceIt("should be consumable from pods in volume with mappings and Item mode set [NodeConformance]", func() {
+		mode := int32(0400)
+		doProjectedConfigMapE2EWithMappings(f, 0, 0, &mode)
+	})
+
+	/*
+	   Release : v1.9
+	   Testname: Projected Volume, ConfigMap, mapped, non-root user
+	   Description: A Pod is created with projected volume source ‘ConfigMap’ to store a configMap as non-root user with uid 1000. The ConfigMap is also mapped to a custom path. Pod MUST be able to read the content of the ConfigMap from the custom location successfully and the mode on the volume MUST be -r-—r——r—-.
+	*/
+	framework.ConformanceIt("should be consumable from pods in volume with mappings as non-root [NodeConformance]", func() {
+		doProjectedConfigMapE2EWithMappings(f, 1000, 0, nil)
+	})
+
+	It("should be consumable from pods in volume with mappings as non-root with FSGroup [NodeFeature:FSGroup]", func() {
+		doProjectedConfigMapE2EWithMappings(f, 1000, 1001, nil)
+	})
+
+	/*
+	   Release : v1.9
+	   Testname: Projected Volume, ConfigMap, update
+	   Description: A Pod is created with projected volume source ‘ConfigMap’ to store a configMap and performs a create and update to new value. Pod MUST be able to create the configMap with value-1. Pod MUST be able to update the value in the confgiMap to value-2.
+	*/
+	framework.ConformanceIt("updates should be reflected in volume [NodeConformance]", func() {
```

Comment:
This test is a subset of the next test.  I think we can remove this test.

Reply:
So optional and non optional projected volumes have exactly the same behavior?

Reply:
Ah I didn't realize there was actually an `Optional` field in the API.  I will add the test back.  Maybe we should add one more test case to test pod failure when optional == false

Reply:
One more observation: non-optional is not tested for secrets either.  So some future areas to improve coverage:
- non-optional for secrets
- non-optional should fail for secrets when key doesn't exist
- non-optional should fail for configmaps when key doesn't exist